roles/download/defaults/main.yml

```diff
@@ -60,7 +60,7 @@ downloads:
     url: "{{ calico_cni_download_url }}"
     owner: "root"
     mode: "0755"
-    enabled: "{{ kube_network_plugin == 'calico' or kube_network_plugin == 'canal' }}"
+    enabled: "{{ kube_network_plugin == 'calico' and not use_calico_cni_container|default(false) }}"
```

Comment:
This may break CI/automation built around calico using kargo. If someone is using binaries to download/install CNI calico plugins, it will not work anymore. Are you sure you want to drop this?